marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -79,11 +82,21 @@ object Api {
 
   private[this] case class AppRsp(app: Option[App] = None) {
 
-    def toAddresses: Set[Address] =
+    private[this] def healthy(healthCheckResults: Seq[HealthCheckResult]): Boolean =
+      healthCheckResults.forall(
+        _ match {
+          case HealthCheckResult(Some(alive)) if (alive) => true
+          case _ => false
+        }
+      )
+
+    private[v2] def toAddresses(marathonHealth: Boolean): Set[Address] =
       app match {
         case Some(App(_, Some(tasks))) =>
           tasks.collect {
-            case Task(_, Some(host), Some(Seq(port, _*))) =>
+            case Task(_, Some(host), Some(Seq(port, _*)), _) if (!marathonHealth) =>
```

Comment:
This is nice.  Better than my filter suggestion.

Style nit: don't need the parens around the if predicate, I don't think.